### ISSUE TITLE
feat(skill): tax-loss harvesting skill + tax-lot data model + wash-sale compliance rule

### ISSUE THREAD

**Issue by srinandan** [srinandan/portfolio-copilot#337]
**Category:** Enhancement (new skill) · **Priority:** P2 · **Effort:** ~1 week · **Labels:** `enhancement`

## Summary

Add a **tax-loss harvesting (TLH)** skill: identify positions held at an unrealized loss in **taxable** accounts, draft a compliant harvest trade (`ProposedAction`), and enforce the **wash-sale rule** as a new deterministic Reviewer check. Inspired by domain logic in Anthropic's financial skills, but built native to this repo's manifest + deterministic-primitive + Reviewer-rule patterns.

## Why this fits the architecture

A TLH skill **`produces: [proposed_action]`** — the same artifact `action-drafting` produces — so the entire existing compliance spine fires with **zero new gate wiring**:

- `reviewer` already carries `mandatory_if: proposed_action exists` → it auto-reviews the harvest trade.
- The HITL approval card and Alpaca execution gate are triggered structurally by that same artifact.
- Routing is **additive** per [ADR-0022](../blob/main/docs/adr/0022-intent-driven-skill-planning.md): shipping `skills/tax-loss-harvesting/` with a `manifest.json` means the planner picks it up with **no orchestrator edit**. This is a real test of the ADR-0022 "drop a skill and go" claim at N=7 skills.

Consistent with [ADR-0016](../blob/main/docs/adr/0016-deterministic-primitives-in-orchestrator.md): the loss identification, lot selection, and wash-sale math are **deterministic** in the orchestrator; the LLM writes only the `rationale`.

## The real prerequisite: a tax-lot / cost-basis data model (bulk of the effort)

TLH is **impossible to do correctly on today's data.** `Position` (`orchestrator/src/orchestrator/contracts/holdings.py`) is `ticker / quantity / asset_class / market_value_usd / account_type` — there is **no cost basis and no acquisition date**. You cannot compute an unrealized loss without cost basis, and you cannot check the wash-sale ±30-day window without purchase dates.

One hook already exists: `account_type` (`taxable` / `retirement`). TLH applies **only to taxable accounts**, and that field is already present to gate on.

So the foundational step is a **tax-lot model** (per-lot: `acquired_at`, `cost_basis_usd`, `quantity`), a `/schemas` schema, the Go + Pydantic contract pair, the `sync-schemas.sh` drift check, and an ingestion path to populate it. This is a **data-model change, not a skill**, and it is most of the work.

## The demo centerpiece: wash-sale as a deterministic compliance rule

This is what makes TLH worth doing *for this project specifically*. The **wash-sale rule** (a loss is disallowed if a substantially-identical security is bought within ±30 days of the sale) maps exactly onto the existing deterministic-compliance pattern. It becomes a new `rule_wash_sale` in `orchestrator/src/orchestrator/reviewer/rules.py`, sitting next to `rule_excluded_ticker`: the LLM proposes a harvest, and a deterministic rule independently blocks any wash-sale violation before it reaches the human. That is a **governance** narrative — the entire point of the demo — and arguably a third demo centerpiece alongside live-revocation and adversarial-resilience.

### Phase 1 — Tax-lot data model (foundation, no TLH behaviour yet)
- [ ] Define `tax-lot.schema.json` under `/schemas` (per-lot: `lot_id`, `ticker`, `acquired_at`, `quantity`, `cost_basis_usd`, `account_type`).
- [ ] Extend the holdings model to carry lots: add an optional `lots: list[TaxLot]` to `Position` (or a sibling `TaxLotSnapshot`), keeping the existing aggregate `Position` fields backward-compatible so nothing else breaks.
- [ ] Author the Go contract (`pkg/contracts/`) and Pydantic contract (`orchestrator/src/orchestrator/contracts/`) as a matched pair; wire the CI drift check (`scripts/sync-schemas.sh`).
- [ ] Extend document ingestion ([ADR-0020](../blob/main/docs/adr/0020-typed-document-ingestion-and-onboarding-prefill.md)) to accept lots in the holdings JSON snapshot; add a `testdata/` fixture with realistic taxable lots (some at a loss, one inside a wash-sale window).
- [ ] Tests: schema round-trips; a lotless snapshot still loads (backward compatibility).

### Phase 2 — Deterministic TLH primitive
- [ ] `orchestrator/src/orchestrator/primitives/tax_loss_harvesting.py::find_harvest_candidates(holdings, lots, ips)` — mirrors the shape of `calculate_draft_action`:
  - filter to `account_type == taxable`;
  - per lot, unrealized loss = `market_value - cost_basis` (loss only); rank by harvestable loss magnitude;
  - flag short-term vs long-term (from `acquired_at`);
  - respect existing IPS constraints (`excluded_tickers`/`excluded_sectors`, concentration) by reusing the existing checks;
  - return the candidate harvest trade dict, or `None` when nothing is worth harvesting (empty is a valid, correct output — never invent a trade).
- [ ] Unit tests: hand-calculated harvestable loss; no-taxable-lots → `None`; gains-only → `None`; short/long-term flagging.

### Phase 3 — The skill
- [ ] `skills/tax-loss-harvesting/SKILL.md` — purpose, inputs, deterministic harvest logic, failure modes, tools/permissions (**no** Alpaca order credential, per [ADR-0005](../blob/main/docs/adr/0005-managed-agents-hybrid-evaluation.md)), acceptance criteria. Mirror `skills/action-drafting/SKILL.md`.
- [ ] `skills/tax-loss-harvesting/manifest.json`:
  ```json
  {
    "id": "tax-loss-harvesting",
    "summary": "Identifies taxable positions held at an unrealized loss and drafts a wash-sale-safe harvest trade.",
    "applies_when": "The user asks to harvest losses, reduce taxes, offset gains, or realize a loss for tax purposes.",
    "requires": ["holdings", "active_ips", "tax_lots"],
    "produces": ["proposed_action"],
    "parallelizable": false
  }
  ```
  (Add a `TAX_LOTS` artifact to the `Artifact` enum in `orchestrator/src/orchestrator/skills/manifest.py`; classify it in `PRELOADED_ARTIFACTS` since it is ingested, not skill-produced.)
- [ ] `skills/tax-loss-harvesting/tax_loss_harvesting.evalset.json`.
- [ ] Register the skill (`scripts/register_skill.sh`) and add to `scripts/register_all_skills.sh`.

### Phase 4 — Wash-sale compliance rule
- [ ] Add `rule_wash_sale` to `orchestrator/src/orchestrator/reviewer/rules.py`: **fails** if a `proposed_action` SELL harvests a loss on a ticker for which any lot of a substantially-identical security was acquired within ±30 days of the sell. Register it in `RULES`.
- [ ] Mirror the rule in `skills/reviewer/SKILL.md`'s "Rules to evaluate" section and the reviewer evalset — the rules module header requires both sides to move together.
- [ ] Tests: harvest inside a wash-sale window → fail (auto-flag to human); harvest outside the window → pass.

### Phase 5 — Planning + UI + docs
- [ ] Add a harvest-intent include rule to `orchestrator/src/orchestrator/planning/intent.py` (verbs: `harvest`, `tax loss`, `offset gains`, `realize a loss`) and an eval fixture `prompt → [tax-loss-harvesting]`.
- [ ] The harvest proposal flows through the existing `ApprovalCard`; add a wash-sale line to the Policy Safety Checklist in the frontend.
- [ ] `docs/adr/00xx-tax-loss-harvesting.md` — the skill, the tax-lot data model, and wash-sale as a compliance rule. Update `CHANGELOG.md` under `[Unreleased]`.

## Acceptance criteria

1. A taxable position at an unrealized loss produces a harvest `ProposedAction`; a lossless or retirement-only portfolio produces zero.
2. Harvestable loss and short/long-term status are computed deterministically in the primitive and verified against hand-calculated values.
3. `rule_wash_sale` blocks (routes to human) any harvest that violates the ±30-day window; the deterministic rule and the reviewer SKILL.md/evalset stay in sync.
4. The skill is routed by the planner from a harvest-intent prompt with **no** change to `planner.py` — only its manifest, the intent rule, and the new `TAX_LOTS` artifact.
5. TLH holds **no** Alpaca order-placement credential; execution still passes through Reviewer + HITL, unchanged.
6. `sync-schemas.sh` drift check passes for the new tax-lot contract across Go and Pydantic; a lotless holdings snapshot remains loadable (backward compatible).

## Open questions

1. **Lot storage** — extend `Position` with an optional `lots` list vs. a separate `TaxLotSnapshot` collection in Firestore.
2. **"Substantially identical"** — v1 scope: exact-ticker wash-sale match only, or also a configurable set of substantially-identical tickers (e.g. `VOO`/`IVV`/`SPY`)? Recommend exact-ticker for v1, note the ETF-equivalence case as follow-up.
3. **Replacement-buy suggestion** — does the skill only propose the loss-realizing sell, or also suggest a non-identical replacement to maintain market exposure? Recommend sell-only for v1 (keeps it one clean `ProposedAction`).

## Sequencing note

TLH is a "grow the catalog" feature and the one that would actually *earn* semantic-search retrieval later. It is larger than the two ~1-day governance fixes (#173 Reviewer re-check after HITL edit, #169 per-session cost cap). Recommend closing **#173 first** — the edit-after-review gap is exactly the bug a wash-sale-sensitive harvest trade would most want closed — then building TLH on solid ground.
